cpp/locality/AABBQuery.cc

```diff
@@ -275,15 +279,16 @@ NeighborBond AABBQueryIterator::next()
                 std::sort(m_current_neighbors.begin(), m_current_neighbors.end());
                 break;
             }
-            else if ((m_r_cur >= m_r_max) || (m_r_cur >= max_plane_distance)
-                     || ((m_all_distances.size() - m_query_points_below_r_min.size()) >= m_num_neighbors))
+
+            if ((m_r_cur >= m_r_max) || (m_r_cur >= max_plane_distance)
+                || ((m_all_distances.size() - m_query_points_below_r_min.size()) >= m_num_neighbors))
             {
                 // Once this condition is reached, either we found enough
                 // neighbors beyond the normal min_plane_distance
                 // condition or we conclude that there are not enough
                 // neighbors left in the system.
                 for (std::map<unsigned int, float>::const_iterator it(m_all_distances.begin());
```

Comment:
You can probably use `auto it(m_all_distances.cbegin())`? Just get a const iterator with `cbegin` / `cend`.
```suggestion
                for (auto it(m_all_distances.cbegin());
```

Reply:
At that point this can probably just use a range-based loop.